FilesCheck.py

```diff
@@ -379,6 +379,25 @@ def py_demarshal_long(b):
     return (b[0] + (b[1] << 8) + (b[2] << 16) + (b[3] << 24))
 
 
+def pyc_magic_from_chunk(chunk):
+    """From given chunk (beginning of the file), return Python magic number"""
+    return py_demarshal_long(chunk[:4]) & 0xffff
+
+
+def pyc_mtime_from_chunk(chunk):
+    """From given chunk (beginning of the file), return mtime or None
+
+    From Python 3.7, mtime is not always present.
+    """
+    magic = pyc_magic_from_chunk(chunk)
+    second = py_demarshal_long(chunk[4:8])
+    if magic >= 3390:  # Python 3.7
```

Comment:
Maybe you could add a link to the specification, i.e. https://www.python.org/dev/peps/pep-0552/#specification

Reply:
added